docs/tutorials/developer-journey/level-2/2.3-advanced-canister-calls.md

```diff
@@ -0,0 +1,252 @@
+
+Certified variables utilize **chain-key cryptography** to generate digital signatures that can then be validated using a single, permanent public key that belongs to the IC. This is a bit different from traditional signatures, since the private key never exists in a single location; it is constantly distributed between many different nodes. Valid signatures can only be generated when the majority of the nodes storing the pieces of the private key cooperate in a cryptographic protocol. Through this method, an application can immediately validate all certified variables that it receives from a query call without having to put trust into the particular node that it received the response from. 
+
+Certification happens at the canister level. Certified variables can be set during an update call to a canister. Recall that an update call changes the canister's state and goes through consensus. When a certified variable is set, the certification can then be read in future query calls and return the certified variable in response to the query call in a trustworthy and secure manner. 
+
+If you want to dive a bit deeper, you can take a look at an the DFINITY example project using certified variables [here](https://github.com/dfinity/examples/tree/master/motoko/cert-var), or view a community example of an application using certified variables [here](https://github.com/nomeata/motoko-certified-http/blob/master/main.mo).
+
+## Composite queries
+
+A composite query call is a type of query that can only be invoked via an ingress message, such as one generated by an agent such as a web browser, or through `dfx`. Composite queries can be used for scenarios such as frontend code that wants to get information from the backend code. A great example of this is a partitioned key-value store setup, where there is one frontend canister and several different backend canisters, with each backend representing a partition of the key-value stores. Composite queries can be very useful by providing efficient data retrieval for dapps that scale vertically with data partitioned across multiple canisters.
+
+:::info
+Composite queries are supported in `dfx` versions `v0.15.0 and newer`. 
+:::
+
+If you want to dive a bit deeper, you can read more on the developer blog [here](https://internetcomputer.org/blog/features/composite-query) or you can take a look at the DFINITY example project using composite queries [here](https://github.com/dfinity/examples/tree/master/motoko/cert-var). 
+
+
+## Inter-canister calls
+
+Inter-canister calls refer to the ability to make calls between different canisters. This functionality is crucial for developers building complex dapps, as it enables you to reuse and share functionality between multiple canisters within the dapp. 
```

Comment:
This framing might be a bit misleading. The general advice we give to developers is to build their dapp in a single canister if possible because managing multiple canisters is an overhead and you need to handle error cases of inter-canister calls, since there's no atomicity anymore if inter-canister calls are involved.

I think a better example of using inter-canister calls is to interact with third-party canisters like fungible and non-fungible tokens or, in general, whenever you have dedicated functionality that might be reused by many different services.